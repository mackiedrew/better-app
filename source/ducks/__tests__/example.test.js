import example, {
  initialState,
  ADD,
  SUBTRACT,
  RESET,
  addExample,
  subtractExample,
  resetExample,
} from "../../ducks/example"

const expectSnapshot = thing => expect(thing).toMatchSnapshot()

describe("example reducer", () => {
  it("should handle no action", () => expectSnapshot(example(initialState)))
  it("should handle empty action", () => expectSnapshot(example(initialState, {})))
  it("should handle ADD", () => expectSnapshot(example(initialState, { type: ADD })))
  it("should handle SUBTRACT", () => expectSnapshot(example(initialState, { type: SUBTRACT })))
  it("should handle RESET", () => expectSnapshot(example(initialState, { type: RESET })))
})

describe("example action creators", () => {
  it("addExample", () => expectSnapshot(addExample()))
  it("subtractExample", () => expectSnapshot(subtractExample()))
  it("resetExample", () => expectSnapshot(resetExample()))
})
