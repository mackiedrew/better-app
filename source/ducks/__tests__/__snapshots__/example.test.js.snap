// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example action creators addExample 1`] = `
Object {
  "type": "example/ADD",
}
`;

exports[`example action creators resetExample 1`] = `
Object {
  "type": "example/RESET",
}
`;

exports[`example action creators subtractExample 1`] = `
Object {
  "type": "example/SUBTRACT",
}
`;

exports[`example reducer should handle ADD 1`] = `
Object {
  "value": 1,
}
`;

exports[`example reducer should handle RESET 1`] = `
Object {
  "value": 0,
}
`;

exports[`example reducer should handle SUBTRACT 1`] = `
Object {
  "value": -1,
}
`;

exports[`example reducer should handle empty action 1`] = `
Object {
  "value": 0,
}
`;

exports[`example reducer should handle no action 1`] = `
Object {
  "value": 0,
}
`;
